src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -182,15 +191,31 @@ func (mod *mgmtModule) handleGetAttachInfo(ctx context.Context, reqb []byte, pid
 		return nil, err
 	}
 
-	mod.monitor.RegisterProcess(ctx, pid)
-
 	return cacheResp, err
 }
 
+func (mod *mgmtModule) handlePoolConnect(ctx context.Context, reqb []byte, pid int32) ([]byte, error) {
+	pbReq := new(mgmtpb.PoolMonitorReq)
+	if err := proto.Unmarshal(reqb, pbReq); err != nil {
+		return nil, drpc.UnmarshalingPayloadFailure()
+	}
+	mod.monitor.RegisterPool(ctx, pid, pbReq)
+	return nil, nil
+}
+
```

Comment:
Same comment as handlePoolConnect re: the return values.

Reply:
same as above